examples/xds-management-server/pkg/k8s/k8s.go

```diff
@@ -0,0 +1,52 @@
+package k8s
+
+import (
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	//  load the gcp plugin (required to authenticate against GKE clusters from outside the cluster)
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+// GetK8sConfig finds and returns a kubernetes config
+func GetK8sConfig(logger *log.Logger, kubeHost, kubeConfigPath string) (*rest.Config, error) {
```

Comment:
Fairly sure https://pkg.go.dev/k8s.io/client-go/tools/clientcmd will do all of this 😃 (see the example at the top) - although it uses KUBECONFIG as the env var, which is likely [what you want anyay](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#set-the-kubeconfig-environment-variable)